src/VecSim/algorithms/raft_ivf/ivf_tiered.h

```diff
@@ -0,0 +1,246 @@
+#pragma once
+
+#include <mutex>
```

Comment:
`<algorithm>` is missing and it messes up with `std::set_union` and `std::copy_n`
```suggestion
#include <mutex>
#include <algorithm>
```